services-core/src/gas_price/priority.rs

```diff
@@ -1,14 +1,36 @@
 use super::GasPriceEstimating;
 use anyhow::{anyhow, Result};
-use std::{future::Future, time::Duration};
+use std::{
+    future::Future,
+    sync::atomic::{AtomicU64, Ordering},
+    time::Duration,
+};
+
+// Errors of an individual estimator are logged as warnings until it has failed this many times in
+// a row at which point they are logged as errors.
+// This is useful to reduce alerts for estimators that sometimes fail individual requests while
+// still getting them when the estimator really goes down.
+const LOG_ERROR_AFTER_N_ERRORS: u64 = 10;
 
 // Uses the first successful estimator.
 pub struct PriorityGasPrice {
-    estimators: Vec<Box<dyn GasPriceEstimating>>,
+    estimators: Vec<Estimator>,
+}
+
+struct Estimator {
+    estimator: Box<dyn GasPriceEstimating>,
+    errors_in_a_row: AtomicU64,
```

Comment:
Not that it makes any difference at all (in fact, I don't think its worth changing), buy I think it would be every so slightly more correct if this were a `AtomicUsize` so it would work better on 32-bit platforms (which we don't compile for :see_no_evil:). Also, this most probably doesn't need 64-bits for counting errors, and the platform's word size is almost certainly enough to count these errors.